include/tvm/relay/type.h

```diff
@@ -298,6 +298,8 @@ class TypeReporterNode : public Node {
   // solver is not serializable.
   void VisitAttrs(tvm::AttrVisitor* v) final {}
 
+  mutable NodeRef location;
```

Comment:
this needs a bit more documentation, especially given that this is a mutable field if we are using it to record context info, we can also consider virtual functions

Reply:
move as a method, put the field in the subclass